cmd/gwi-api/api/stores/postgrestore/store.go

```diff
@@ -0,0 +1,198 @@
+package postgrestore
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/tchap/gwi-challenge/cmd/gwi-api/api"
+
+	"github.com/jmoiron/sqlx"
+	"github.com/lib/pq"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+)
+
+// Store implements api.Store while keeping all data Postgres.
+// pqcrypto must be enabled on the database being used.
+type Store struct {
+	logger *zap.Logger
+	db     *sqlx.DB
+}
+
+// New returns a new Store instance using the given database.
+func New(logger *zap.Logger, db *sqlx.DB) *Store {
+	return &Store{
+		logger: logger,
+		db:     db,
+	}
+}
+
+func (s *Store) AuthenticateOrCreateVolunteerAccount(ctx context.Context, volunteer *api.Volunteer) error {
```

Comment:
Do you have any experience with some ORMs?

Reply:
We are actually using http://gorm.io on the project I am on currently, but I am really not convinced about the added value. ORM is fine for simple cases, here it could be used for sure, but then it may start getting under your hand too much. And I actually like to know what query I am sending to the DB server 🙂

Reply:
We have exactly the same approach 😄